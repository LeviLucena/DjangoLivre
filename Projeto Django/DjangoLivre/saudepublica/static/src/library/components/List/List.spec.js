import { shallowMount } from '@vue/test-utils'
import BrItem from '../Item/Item.ce.vue'
import BrList from './List.ce.vue'

describe('br-list', () => {
  test('Testa se a classe br-list está renderizando.', () => {
    const wrapper = shallowMount(BrList)
    expect(wrapper.find('.br-list').exists()).toBe(true)
  })

  test('Testa se a prop horizontal está renderizando a classe respectiva.', () => {
    const wrapper = shallowMount(BrList, {
      propsData: { horizontal: true },
    })
    expect(wrapper.find('.horizontal').exists()).toBe(true)
  })

  test('Testa se a prop title é exibida.', () => {
    const titleExample = 'title example'
    const wrapper = shallowMount(BrList, {
      propsData: {
        title: titleExample,
      },
    })
    expect(wrapper.text()).toMatch(titleExample)
  })

  test('Testa se a prop data-unique foi acionada.', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        dataUnique: true,
      },
    })
    expect(wrapper.attributes('data-unique')).toBe('true')
  })

  test('Testa se a prop data-toggle foi acionada.', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        dataToggle: true,
      },
    })
    expect(wrapper.attributes('data-toggle')).toBe('true')
  })

  test('Inclui item no slot default', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        title: 'Título do List',
      },
      slots: {
        default: `
          <br-item title="Título do Item">
          </br-item>`,
      },
      global: {
        stubs: {
          'br-item': BrItem,
        },
      },
    })
    expect(wrapper.findComponent(BrItem))
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set density attribute small', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        density: 'small',
        title: 'Título do List',
      },
      slots: {
        default: `<br-item title="Título do Item"></br-item>`,
      },
      global: {
        stubs: {
          'br-item': BrItem,
        },
      },
    })
    expect(wrapper.find('.py-4').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set density attribute medium', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        density: 'medium',
        title: 'Título do List',
      },
      slots: {
        default: `<br-item title="Título do Item"></br-item>`,
      },
      global: {
        stubs: {
          'br-item': BrItem,
        },
      },
    })
    expect(wrapper.find('.py-3').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set density attribute large', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        density: 'large',
        title: 'Título do List',
      },
      slots: {
        default: `<br-item title="Título do Item"></br-item>`,
      },
      global: {
        stubs: {
          'br-item': BrItem,
        },
      },
    })
    expect(wrapper.find('.py-3').exists()).toBe(false)
    expect(wrapper.find('.py-4').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('if the hideTitleDivider property hides the divider', () => {
    const wrapper = shallowMount(BrList, {
      propsData: {
        hideTitleDivider: true,
        title: 'Título do List',
      },
      slots: {
        default: `<br-item title="Título do Item"></br-item>`,
      },
      global: {
        stubs: {
          'br-item': BrItem,
        },
      },
    })
    expect(wrapper.find('.br-divider').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })
})
