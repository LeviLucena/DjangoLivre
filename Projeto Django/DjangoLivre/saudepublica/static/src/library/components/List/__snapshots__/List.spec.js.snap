// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`br-list Inclui item no slot default 1`] = `
<div
  class="br-list"
  data-toggle="false"
  data-unique="false"
  density="large"
  role="list"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Título do List
    </div>
  </div>
  <br-divider-stub
    dashed="false"
    vertical="false"
  />
  
  <div
    class="item-wrapper"
  >
    <div
      class="br-item align-items-center"
    >
      <!--v-if-->
      <span>
        
        
      </span>
    </div>
    <span
      style="display: none;"
    >
      
      
    </span>
  </div>
  
</div>
`;

exports[`br-list if the hideTitleDivider property hides the divider 1`] = `
<div
  class="br-list"
  data-toggle="false"
  data-unique="false"
  density="large"
  role="list"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Título do List
    </div>
  </div>
  <!--v-if-->
  
  <div
    class="item-wrapper"
  >
    <div
      class="br-item align-items-center"
    >
      <!--v-if-->
      <span>
        
        
      </span>
    </div>
    <span
      style="display: none;"
    >
      
      
    </span>
  </div>
  
</div>
`;

exports[`br-list set density attribute large 1`] = `
<div
  class="br-list"
  data-toggle="false"
  data-unique="false"
  density="large"
  role="list"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Título do List
    </div>
  </div>
  <br-divider-stub
    dashed="false"
    vertical="false"
  />
  
  <div
    class="item-wrapper"
  >
    <div
      class="br-item align-items-center"
    >
      <!--v-if-->
      <span>
        
        
      </span>
    </div>
    <span
      style="display: none;"
    >
      
      
    </span>
  </div>
  
</div>
`;

exports[`br-list set density attribute medium 1`] = `
<div
  class="br-list"
  data-toggle="false"
  data-unique="false"
  density="medium"
  role="list"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Título do List
    </div>
  </div>
  <br-divider-stub
    dashed="false"
    vertical="false"
  />
  
  <div
    class="item-wrapper"
  >
    <div
      class="br-item align-items-center py-3"
    >
      <!--v-if-->
      <span>
        
        
      </span>
    </div>
    <span
      style="display: none;"
    >
      
      
    </span>
  </div>
  
</div>
`;

exports[`br-list set density attribute small 1`] = `
<div
  class="br-list"
  data-toggle="false"
  data-unique="false"
  density="small"
  role="list"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Título do List
    </div>
  </div>
  <br-divider-stub
    dashed="false"
    vertical="false"
  />
  
  <div
    class="item-wrapper"
  >
    <div
      class="br-item align-items-center py-4"
    >
      <!--v-if-->
      <span>
        
        
      </span>
    </div>
    <span
      style="display: none;"
    >
      
      
    </span>
  </div>
  
</div>
`;
